reposerver/repository/repository.go

```diff
@@ -1721,6 +1721,15 @@ func populateHelmAppDetails(res *apiclient.RepoAppDetailsResponse, appPath strin
 
 	if q.Source.Helm != nil {
 		selectedValueFiles = q.Source.Helm.ValueFiles
+
+		for i, file := range selectedValueFiles {
+			// update path of value file if value file is referencing another ApplicationSource
+			if strings.HasPrefix(file, "$") {
+				pathStrings := strings.Split(file, "/")
+				pathStrings[0] = os.Getenv(strings.Split(file, "/")[0])
+				selectedValueFiles[i] = strings.Join(pathStrings, "/")
+			}
+		}
```

Comment:
I don't think this actually allows referencing an external repo. `strings.Split(file, "/")[0]` is always going to return `$something`, and `os.Getenv` is always going to return an empty string (because no env var in repo-server starts with a `$`). So the final values file path will just be `file` with `$something` removed from the beginning. The code effectively can only reference files from the same repo.